docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -44,6 +44,8 @@ See the discussion below for background and for detailed discussion of each opti
 
 Developers should ensure that their programs compile and pass their automated tests with all these options, e.g., by setting these as the default options. We encourage developers to consider it a bug if the program cannot be compiled with these options. Those who build programs for production may choose to omit some options that hurt performance if the program only processes trusted data, but remember that it's not helpful to deploy programs that that are insecure and rapidly do the wrong thing. Existing programs may need to be modified over time to work with some of these options.
 
+*Production* code is code intended for use in the real world with real effects; it should be maximally reliable and performant. *Instrumented test* code is code that is instrumented to improve defect detection and debuggability, and as such, often crashes more and is slower. Test processes should use both instrumented test code and production code.
```

Comment:
Should the new paragraph go before the previous paragraph since that already refers to "*production code*"?

The last sentence and in the new paragraph and the first sentence in the previous paragraph also seem to partly overlap. Maybe these could be combined for clarity:

> In this guide, we refer to *production code* for code intended for use in the real world with real effects; it should be maximally reliable and performant. We refer to *instrumented test code* for code that is instrumented to improve defect detection and debuggability, and as such, often crashes more and is slower. 
>
> Developers should ensure that both their production code and their instrumented test code pass their automated test with all these options. We encourage developers to consider it a bug if the program cannot be compiled with these options. Those who build production code may choose to omit some options that hurt performance if the program only processes trusted data, but remember that it's not helpful to deploy programs that that are insecure and rapidly do the wrong thing. Existing programs may need to be modified over time to work with some of these options.

Thoughts on this?

Reply:
Good idea. I'll tweak that text a little & create a new PR.